src/Explorer/Tabs/TabsManager.ts

```diff
@@ -4,23 +4,25 @@ import Explorer from "../Explorer";
 import TabsBase from "./TabsBase";
 
 export class TabsManager {
-  public openedTabs: ko.ObservableArray<TabsBase>;
-  public activeTab: ko.Observable<TabsBase>;
+  private idx = ko.observable(0);
+  public openedTabs = ko.observableArray<TabsBase>([]);
+  public activeTab = ko.pureComputed(() => this.openedTabs()[this.idx()]);
 
   constructor() {
-    this.openedTabs = ko.observableArray<TabsBase>([]);
-    this.activeTab = ko.observable<TabsBase>();
+    this.openedTabs.subscribe(
+      (log) => log.forEach(({ value: tab, status }) => tab.manager(status === "deleted" ? undefined : this)),
```

Comment:
Extra log left behind?

Reply:
No, it's the array change*log*. I'll give it a better name.